__tests__/app.test.js

```diff
@@ -60,8 +60,105 @@ describe('/api/articles', () => {
             expect(body.allArticles).toBeSortedBy('created_at', { descending: true})
         });
     });
+
+    test('TOPIC = filter: filtered array of topics retrieved', () => {
+        return request(app)
+        .get('/api/articles?topic=mitch')
+        .expect(200)
+        .then(({body}) => {
+            expect(body.allArticles.length).toBe(11);
+            expect(body.allArticles.every((article) => article.topic === 'mitch')).toBe(true);
+        })
+    });
+
+    test('TOPIC = filter: empty array retrieved if topic has no associated articles', () => {
+        return request(app)
+        .get('/api/articles?topic=paper')
+        .expect(200)
+        .then(({body}) => {
+            expect(Array.isArray(body.allArticles)).toBe(true); 
+            expect(body.allArticles.length).toBe(0);        
+        })
+    });
+
+    test('TOPIC = filter: 404 if non-existent topic requested', () => {
+        return request(app)
+        .get('/api/articles?topic=pelican')
+        .expect(404)
+        .then(({body}) => {
+            expect(body.msg).toBe('Topic not found!')      
+        })
+    });
+
+    test('SORT - array sorted by (string) title', () => {
+        return request(app)
+        .get('/api/articles?sort_by=title')
+        .expect(200)
+        .then(({body}) => {
+            expect(body.allArticles.length).toBe(12);
+            expect(body.allArticles).toBeSortedBy('title', { descending: true})
+        })
+    });
+
+    test('SORT - array sorted by (integer) article_id', () => {
+        return request(app)
+        .get('/api/articles?sort_by=article_id')
+        .expect(200)
+        .then(({body}) => {
+            expect(body.allArticles.length).toBe(12);
+            expect(body.allArticles).toBeSortedBy('article_id', { descending: true})
+        })
+    });
+
+    test('SORT - array sorted by (integer) comment_count', () => {
+        return request(app)
+        .get('/api/articles?sort_by=comment_count')
+        .expect(200)
+        .then(({body}) => {
+            console.log(body.allArticles)
+            expect(body.allArticles.length).toBe(12);
+            expect(body.allArticles).toBeSortedBy('comment_count', { descending: true})
+        })
+    });
+
+    test('SORT - if sort_by column does not exist, defaults to standard [created_at]', () => {
+        return request(app)
+        .get('/api/articles?sort_by=pelican')

```

Comment:
Nice logic, but this should be an error. Otherwise, there would be no way to understand that the expected sort_by had not been performed